### PR TITLE
[Enhancement] compatible with old cn

### PR DESCRIPTION
**Pull request by abc982627271**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 4 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|4|00.00%|[157, 160, 161, 162]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/service/service_be/starrocks_be.cpp [hunk 1]
```diff
@@ -153,6 +153,14 @@ void start_be(const std::vector<StorePath>& paths, bool as_cn) {
...
     // Start thrift server
     auto thrift_server = BackendService::create<BackendService>(exec_env, config::be_port);
+    if (as_cn) {
+        int thrift_port = config::thrift_port;
+        if (thrift_port != 0) {
+            thrift_server = BackendService::create<BackendService>(exec_env, thrift_port);
+            LOG(WARNING) << "thrif_port has been deprecated, please use be_port next time!";
+        }
+    }
+
```
Comment (wyb) [inline be/src/service/service_be/starrocks_be.cpp:163]: int thrift_port = config::be_port;
if (as_cn && config::thrift_port != 0) {
    thrift_port = config::thrift_port;
    LOG(WARNING) << "thrif_port has been deprecated, please use be_port next time!";
}
auto thrift_server = BackendService::create<BackendService>(exec_env, thrift_port);

and the following log is
LOG(ERROR) << "Fail to start BackendService thrift server on port " << thrift_port << ": " << status;

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|5|00.00%|[156, 157, 158, 159, 160]|

File: be/src/service/service_be/starrocks_be.cpp [hunk 2]
```diff
@@ -153,6 +153,14 @@ void start_be(const std::vector<StorePath>& paths, bool as_cn) {
 
     // Start thrift server
     auto thrift_server = BackendService::create<BackendService>(exec_env, config::be_port);
+    if (as_cn) {
+        int thrift_port = config::thrift_port;
+        if (thrift_port != 0) {
+            thrift_server = BackendService::create<BackendService>(exec_env, thrift_port);
+            LOG(WARNING) << "thrif_port has been deprecated, please use be_port next time!";
```
Comment (kevincai) [inline be/src/service/service_be/starrocks_be.cpp:160]:
```suggestion
            LOG(WARNING) << "'thrift_port' has been deprecated, please update be.conf to use 'be_port' instead!";
```

**Review by kevincai — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by kevincai** [comment]
need tick on the parameter change, this is a default parameter change PR.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|6|00.00%|[155, 156, 157, 158, 160, 163]|

**Review by wyb — APPROVED** [review]

**Comment by abc982627271** [comment]
@mergifyio backport branch-3.2

**Comment by abc982627271** [comment]
@mergifyio backport branch-3.1

**Pull request merged**